common/common.go

```diff
@@ -486,3 +487,21 @@ func GetExitStatus(err error) (int, error) {
 	}
 	return -1, err
 }
+
+// GetArch returns the current architecture. It returns runtime.GOARCH if not on a arm device
+// and armv6l, armv7l, arm64 otherwise
+func GetArch() string {
+	arch := runtime.GOARCH
+	if arch == "arm" {
+		utsName := &syscall.Utsname{}
+		syscall.Uname(utsName)
+		arch = ""
+		for _, b := range utsName.Machine {
+			if b == 0 {
+				break
+			}
+			arch += string(b)
+		}
+	}
+	return arch
```

Comment:
Strange, your code works but the result is not as I expected:
```
pi@bb-test-north:~ $ uname -a
Linux bb-test-north 4.4.17-v7+ #904 SMP Tue Aug 16 17:02:00 BST 2016 armv7l GNU/Linux
pi@bb-test-north:~ $ go run arm-test.go 
6
pi@bb-test-north:~ $ GOARM=6 go run arm-test.go 
6
pi@bb-test-north:~ $ GOARM=7 go run arm-test.go 
7
```

Go assumes at compile time that you are going to use a armv6l device (If you dont specify something else), since so the code can be executed on armv6l and armv7l devices. Dave Cheney [says here](https://dave.cheney.net/2015/08/22/cross-compilation-with-go-1-5) that there is not much benefit from compiling especially for armv7l devices.

I don't know if this should be the source of truth. The syscall I made is safer I think.

But by the way, your gist is very smart. I didn't even know about the possibility to instruct the linker to do such things. Is this documented somewhere?